src/Image.h

```diff
@@ -157,7 +157,8 @@ class ImageBase {
     /** Get the address of a particular pixel. */
     void *address_of(int x, int y = 0, int z = 0, int w = 0) const {
         uint8_t *ptr = (uint8_t *)origin;
-        size_t offset = x*stride_0 + y*stride_1 + z*stride_2 + w*stride_3;
+        uint64_t offset = (int64_t)x*stride_0 + (int64_t)y*stride_1 +
```

Comment:
Yes, it should be.

On Fri, Oct 23, 2015 at 11:37 AM, Tyler Denniston <notifications@github.com>
wrote:

> In src/Image.h
> <https://github.com/halide/Halide/pull/952#discussion_r42899089>:
>
> > @@ -157,7 +157,8 @@ class ImageBase {
> >      /** Get the address of a particular pixel. */
> >      void *address_of(int x, int y = 0, int z = 0, int w = 0) const {
> >          uint8_t *ptr = (uint8_t *)origin;
> > -        size_t offset = x*stride_0 + y*stride_1 + z*stride_2 + w*stride_3;
> > +        uint64_t offset = (int64_t)x*stride_0 + (int64_t)y*stride_1 +
>
> I wasn't sure about this: is size_t 64 bits on 64-bit platforms?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/halide/Halide/pull/952/files#r42899089>.
>